apps/designer/app/routes/rest/pages.$projectId.ts

```diff
@@ -1,6 +1,89 @@
-import type { LoaderFunction } from "@remix-run/node";
+import { LoaderFunction, ActionFunction } from "@remix-run/node";
 import { db } from "@webstudio-is/project/index.server";
-import { Pages } from "@webstudio-is/project";
+import { type Pages, utils } from "@webstudio-is/project";
+import { zfd } from "zod-form-data";
+import { z } from "zod";
+import { CreatePageData } from "~/designer/features/sidebar-left/panels/pages/settings";
+import { sentryException } from "~/shared/sentry";
+
+const nonEmptyString = z
+  .string({
+    invalid_type_error: "Can't be empty",
+    required_error: "Can't be empty",
+  })
+  .transform((v) => v.trim())
+  .refine((val) => val !== "", "Can't be empty");
+
+const CreatePageInput = zfd.formData({
+  name: nonEmptyString,
+  path: nonEmptyString
+    .refine((path) => path !== "/", "Can't be just a /")
+    .transform((path) => (path.endsWith("/") ? path.slice(0, -1) : path))
+    .transform((path) => path.replace(/\/\//g, "/"))
+    .refine(
+      (path) => path === "" || path.startsWith("/"),
+      "Must start with a /"
+    )
+    .refine(
+      (path) => /^[-_a-z0-9\\/]*$/.test(path),
+      "Only a-z, 0-9, -, _ and / are allowed"
+    ),
+});
+
+const handlePUT = async (
```

Comment:
Lets use strict camel case without accounting for abbreviations or anything. imageUrl etc. ... makes it much easier to keep consistent and convert automatically when needed
```suggestion
const handlePut = async (
```